testing/tests/DevExpress.ui.widgets/treeViewParts/lazyRendering.js

```diff
@@ -22,6 +28,81 @@ QUnit.test("Only root nodes should be rendered by default", function(assert) {
     assert.equal(items.length, 2);
 });
 
+['!/#$%&\'()"+./:;<=>?@[]^`{|}~\\,', '____2______.jpg', 'E:\\test\\[gsdfgfd]  |  \'[some__file]', '!@#$%^&*()_+'].forEach((testId) => {
+    QUnit.test(`Nodes expanding should work with special charactes in id - ${testId}`, function(assert) {
+        let treeView = createInstance({
+                dataSource: [
+                    { id: testId, text: "item1", selected: false, expanded: false },
+                    { id: testId + 'item1_1', parentId: testId, text: "item1_1", selected: false, expanded: false }
+                ],
+                dataStructure: "plain",
+                height: 500
+            }),
+            parentItem = treeView.getElement().find('[aria-level="1"]'),
+            childItem = treeView.getElement().find('[aria-level="2"]');
+
+        assert.equal(childItem.length, 0);
+
+        treeView.instance.expandItem(parentItem);
+        childItem = treeView.getElement().find('[aria-level="2"]');
+        assert.equal(childItem.length, 1);
+        assert.equal(treeView.hasInvisibleClass(childItem), false);
+    });
+
+    QUnit.test(`Nodes selection should work with special charactes in id - ${testId}`, function(assert) {
+        const treeView = createInstance({
+            dataSource: [
+                { id: testId, text: "item1", selected: false, expanded: true },
+                { id: testId + 'item1_1', parentId: testId, text: "item1_1", selected: false, expanded: true }
+            ],
+            dataStructure: "plain",
+            showCheckBoxesMode: "normal",
+            height: 500
+        });
+
+        treeView.checkSelectedNodes([]);
+
+        const elem = treeView.getElement().find('[aria-level="1"]');
+        treeView.instance.selectItem(elem);
+
+        treeView.checkSelectedNodes([0, 1]);
+    });
+
+    QUnit.test(`Search should work with special charactes in the nodes ids - ${testId}`, function(assert) {
+        let treeView = createInstance({
+                dataSource: [
+                    { id: testId, text: "item1", selected: false, expanded: false },
+                    { id: testId + 'item1_1', parentId: testId, text: "item1_1", selected: false, expanded: false },
+                    { id: 'item2', text: "item2", selected: false, expanded: false }
+                ],
+                dataStructure: "plain",
+                searchEnabled: true,
+                height: 500
+            }),
+            parentItem = treeView.getElement().find('[aria-label="item1"]'),
+            childItem = treeView.getElement().find('[aria-label="item1_1"]'),
+            anotherItem = treeView.getElement().find('[aria-label="item2"]');
+
+        assert.equal(parentItem.length, 1);
+        assert.equal(childItem.length, 0);
+        assert.equal(anotherItem.length, 1);
+
+        const $input = treeView.getElement().find('.dx-texteditor-input');
+        keyboardMock($input).type("1_1");
```

Comment:
The 'type symbols to the search input' approach involves a long list of functions/mocks/wrappers/controls to perform all underlying actions before the first assert occurs.
It doesn't seem to be stable enough with our all our testing configs/envs.
IMHO ~5% of runs will fail without a reason.